### PR TITLE
Initial test suite for Ahoy engine.

### PR DESCRIPTION
**Pull request by krln**
Hello!

I've worked with Ahoy a lot as a part of my Rails Girls Summer of Code work on Species+ project. It's a great gem, but I noticed it could use some tests. So here they are :)

Went with the standard Rails engine testing approach, i.e. "dummy" app stored inside `spec/` which is a minimal Rails app utilizing the engine.

Contains:
- rspec environment with standard dummy app
- test for `Ahoy::EventsController` (one used by javascript api)
- test for `ahoy.track` in Rails app's controller
- test for `Ahoy::VisitsController`
- `.travis.yml` file

I'm not sure if `.travis.yml` is going to run the suite out-of-the-box, it might need an extra `before_script` step for building the dummy app's test database. There's only one way to find out (due to historical reasons .travis.yml needs to be in master first)!

If these tests are not enough or not the way you'd prefer them to be, I'd be happy to learn.

### CODE REVIEW

**Comment by saturnflyer** [comment]
Is there any feedback on this PR? I'm interested in adding tests too and would like to see some guidelines for how to approach them for the project

File: spec/controllers/ahoy/visits_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'spec_helper'
+
+describe Ahoy::VisitsController, :type => :controller do
+  routes { Ahoy::Engine.routes }
+
+  it "creates visits" do
+    begin
+      previous_value = Ahoy.track_visits_immediately = true
```
Comment (gltarsa) [inline spec/controllers/ahoy/visits_controller_spec.rb]: You may want to do this in two steps if your intent is to preserve the old value.  This code is setting both values to true.

**Comment by ankane** [comment]
Hey @krln, thanks for the PR 👍 and really sorry for the (long......) delay. Would love to add tests. I'm thinking we can use something like [Combustion](https://github.com/pat/combustion) so we don't have to include a full Rails app in the project. I know it's been a long time since you put this together, but let me know if you're still interested in helping with it.

**Comment by ankane** [comment]
Cleaning up PRs

**Pull request closed without merging**